sale_import_delivery_carrier/components/importer_sale_channel.py

```diff
@@ -0,0 +1,25 @@
+from odoo.addons.component.core import Component
+
+
+class ImporterSaleChannel(Component):
+    _inherit = "importer.sale.channel"
+
+    def finalize(self, sale_order, raw_import_data):
+        super().finalize(sale_order, raw_import_data)
+        self.create_delivery_line(sale_order, raw_import_data)
+
+    def create_delivery_line(self, sale_order, data):
+        if not data.get("delivery_carrier"):
+            return
+        delivery_carrier = self.env["delivery.carrier"].search(
+            [("name", "=", data["delivery_carrier"]["name"])]
+        )
+        if delivery_carrier:
+            sale_order.carrier_id = delivery_carrier
+            price = data["delivery_carrier"]["price_unit"]
+            discount = data["delivery_carrier"]["discount"]
+            carrier_line = sale_order._create_delivery_line(delivery_carrier, price)
```

Comment:
I think you said we should create the delivery line ourselves manually instead of using that method. Could you confirm ?